Library/Homebrew/diagnostic.rb

```diff
@@ -1121,6 +1121,18 @@ def check_for_external_cmd_name_conflict
         message
       end
 
+      def check_for_beta_xquartz
+        return unless MacOS::XQuartz.version.include? "beta"
+
+        <<-EOS.undent
+        The following beta release of XQuartz is installed: #{MacOS::XQuartz.version}
+
+        XQuartz beta releases include address sanitization, and do not work with
+        all software; notably, wine will not work with beta releases of XQuartz.
+        We recommend only installing stable releases of XQuartz.
+        EOS
```

Comment:
Would you mind indenting the contents between `<<-EOS.undent` and `EOS` by two spaces? That's the style we use in all other messages in this file.